scripts/lib/CIME/case_st_archive.py

```diff
@@ -341,50 +430,35 @@ def _archive_process(case, archive, last_date, archive_incomplete_logs, copy_onl
     """
 
     logger.debug('In archive_process...')
-    compset_comps = case.get_compset_components()
-    compset_comps.append('cpl')
-    compset_comps.append('dart')
 
-    if copy_only is True:
-        archive_file_fn = shutil.copyfile
-    else:
-        archive_file_fn = shutil.move
+    archive_file_fn = _get_archive_file_fn(copy_only)
 
     # archive log files
     _archive_log_files(case, archive_incomplete_logs, archive_file_fn)
 
-    for archive_entry in archive.get_entries():
-        # determine compname and compclass
-        compname, compclass = archive.get_entry_info(archive_entry)
-
-        # check for validity of compname
-        if compname not in compset_comps:
-            continue
-
-        # archive restarts and all necessary associated fields (e.g. rpointer files)
-        logger.info('-------------------------------------------')
-        logger.info('doing short term archiving for {} ({})'.format(compname, compclass))
-        logger.info('-------------------------------------------')
-        datenames = _get_datenames(case, last_date)
-        for datename in datenames:
-            logger.info('Archiving for date %s' % datename)
-            datename_is_last = False
-            if datename == datenames[-1]:
-                datename_is_last = True
-
-            # archive restarts
-            histfiles_savein_rundir = _archive_restarts(case, archive, archive_entry,
-                                                        compclass, compname,
-                                                        datename, datename_is_last,
-                                                        archive_file_fn)
-
-            # if the last datename for restart files, then archive history files
-            # for this compname
-            if datename_is_last:
-                logger.info("histfiles_savein_rundir {} ".format(histfiles_savein_rundir))
-                _archive_history_files(case, archive, archive_entry,
-                                       compclass, compname, histfiles_savein_rundir,
-                                       archive_file_fn)
+    # archive restarts and all necessary associated files (e.g. rpointer files)
```

Comment:
This major refactor could use a careful review. Note that I have reversed the loop order, so that the date loop is now the outer loop. This was needed so that a test could easily call the restart archiving routine for a single date. Doing this refactor also incidentally fixed #1851 and #1853